.devcontainer/Dockerfile

```diff
@@ -333,40 +344,78 @@ RUN apt-key add /tmp/jfrog.pub && \
-RUN pip3 install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
+RUN ${PYTHON_VENV_EXECUTABLE} -m pip install --quiet --upgrade --no-build-isolation --no-cache-dir --verbose --editable .[dev] && \
     rm -rf orc8r.egg-info
 
-# install formatter autopep8
-RUN pip install --no-cache-dir autopep8
+### install formatter autopep8
+RUN ${PYTHON_VENV_EXECUTABLE} -m pip install --no-cache-dir autopep8
 
 #### protos
-ARG PYTHON_LIB=/usr/local/lib/python3.8
+ARG GEN_DIR=lib/python${PYTHON_VERSION}/gen
 
 COPY /protos/ ${MAGMA_ROOT}/protos/
 COPY /lte/protos/ ${MAGMA_ROOT}/lte/protos/
 COPY /orc8r/protos/ ${MAGMA_ROOT}/orc8r/protos/
 COPY /feg/protos/ ${MAGMA_ROOT}/feg/protos/
 COPY /dp/protos/ ${MAGMA_ROOT}/dp/protos/
-WORKDIR $MAGMA_ROOT
-RUN pip install mypy-protobuf && \
-    mkdir $PYTHON_LIB/gen && \
+WORKDIR ${MAGMA_ROOT}
+RUN ${PYTHON_VENV_EXECUTABLE} -m pip install --no-cache-dir "mypy-protobuf==2.4" && \
+    mkdir ${PYTHON_VENV}/${GEN_DIR} && \
     for PROTO_SRC in orc8r lte feg dp; \
     do \
-    python protos/gen_protos.py ${PROTO_SRC}/protos ${MAGMA_ROOT},orc8r/protos/prometheus ${MAGMA_ROOT} $PYTHON_LIB/gen && \
-    python protos/gen_prometheus_proto.py ${MAGMA_ROOT} $PYTHON_LIB/gen; \
+    ${PYTHON_VENV_EXECUTABLE} protos/gen_protos.py ${PROTO_SRC}/protos ${MAGMA_ROOT},orc8r/protos/prometheus ${MAGMA_ROOT} ${PYTHON_VENV}/${GEN_DIR} && \
+    ${PYTHON_VENV_EXECUTABLE} protos/gen_prometheus_proto.py ${MAGMA_ROOT} ${PYTHON_VENV}/${GEN_DIR}; \
     done && \
-    echo "$PYTHON_LIB/gen" > $PYTHON_LIB/dist-packages/magma_gen.pth
+    echo "${PYTHON_VENV}/${GEN_DIR}" > ${PYTHON_VENV}/lib/python${PYTHON_VERSION}/site-packages/magma_gen.pth
+
+### swagger
+ENV SWAGGER_CODEGEN_JAR=/var/tmp/codegen/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar
+ARG MVN_VERSION=3.5.4
+ARG CODEGEN_VERSION=v2.2.3
+
+WORKDIR /var/tmp/
+
+RUN wget http://mirrors.ibiblio.org/apache/maven/maven-3/${MVN_VERSION}/binaries/apache-maven-${MVN_VERSION}-bin.tar.gz && \
+    tar --extract --verbose --file apache-maven-${MVN_VERSION}-bin.tar.gz
+
+RUN git clone --branch ${CODEGEN_VERSION} https://github.com/swagger-api/swagger-codegen /var/tmp/codegen
+
+WORKDIR /var/tmp/codegen
+RUN ../apache-maven-${MVN_VERSION}/bin/mvn clean package -DskipTests
+
+COPY lte/swagger/*.yml /tmp/lte/swagger/
+COPY orc8r/swagger/*.yml /tmp/orc8r/swagger/
+RUN for SWAGGER_SRC in lte orc8r; \
+    do \
+    mkdir -p ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs && \
+    # Copy swagger specs over to the build directory,
+    # so that eventd can access them at runtime
+    cp /tmp/${SWAGGER_SRC}/swagger/* ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs && \
+    # Generate the files
+    ls ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs \
+    | grep -e ".*\.yml" \
+    | xargs -t -I% /usr/bin/java -jar ${SWAGGER_CODEGEN_JAR} generate \
+    -i ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs/% \
```

Comment:
This feels somehow so broken to me, that I cannot resist in making a suggestion to improve.
I would guess, glob patterns come to mind, which should do in most cases.
How about
```bash
    ls ${PYTHON_VENV}/${GEN_DIR}/${SWAGGER_SRC}/swagger/specs/*.yml \
    | xargs -t -I% /usr/bin/java -jar ${SWAGGER_CODEGEN_JAR} generate \
    -i % \
```

Reply:
Thanks. I will look into it!

Reply:
Had to add another pipe to extract the filename (Because `ls` returns the full path of the file). I didn't find an option in `ls` to avoid this. Another option would be to `cd` into the directory first. But I guess we don't want that either.